### ISSUE TITLE
[SUGGESTION] Buffs and buff timers

### ISSUE THREAD

**Issue by SkelemanBaron** [cursey/kanan-new#81]
Seeing as how there's the Status UI Window, I think it'd be nice to be able to see buff durations in that window. Soul Link especially. 

Perhaps cooldowns as well? Or possibly a numeric value on the skill icon?

**Comment by pinkestt** [comment]
I wonder whether remaining duration is stored anywhere. I suppose if it's not there's always the option of storing the start time and total duration (I'm guessing that's stored? Otherwise it could be looked up in a table based on skill+rank) and calculating it from that.

**Comment by Rietty** [comment]
@pinkestt I don't believe it's stored anywhere, but you can do it in a way with a config file set to that person's specific buff duration and simply note start time instead. That's the easiest way to accomplish it, and what the Chinese mod called Tin does.

Calculating based on skill rank won't work because their is a lot of enchant modifiers and reforges that extend buff duration.

**Comment by SkelemanBaron** [comment]
There are some buffs (Again, Soul Link) you can check the remaining duration of in the character window if you hover over the icon, so surely those are stored *somewhere* at the very least.

**Comment by Aahzmandius** [comment]
Cursey was nice enough to add a barebones packet listener branch,
[https://github.com/cursey/kanan-new/tree/alissa-listener](https://github.com/cursey/kanan-new/tree/alissa-listener)

So with a little effort you could listen for the buff packets to start timers and stuff. I started on it a couple months ago, but got distracted adding in a graphics lib to draw skill icons in the status UI. Then work got busy and I haven't messed with it since.

**Comment by cursey** [comment]
Yeah that would probably be the easiest way to implement this. There are some components included in the player entity that might have the information for buffs but reversing them is time consuming and the structures do change fairly often.

**Comment by LoneDev6** [comment]
Could we include this https://github.com/cursey/kanan-new/tree/alissa-listener in the latest version of kanan-new ?

**Comment by cursey** [comment]
I don't see why not. It doesn't really do anything yet though which is why I haven't merged it yet.
